src/core_plugins/vega/public/_vega_vis.scss

```diff
@@ -0,0 +1,132 @@
+
+.vgaVis {
+  display: flex;
+  flex: 1 1 100%;
+  position: relative;
+  // flex-direction determined by js
+}
+
+.vgaVis__view {
+  z-index: 0;
+  flex: 1 1 100%;
+
+  display: block;
+  max-width: 100%;
+  max-height: 100%;
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+
+  // BUG #23514: Make sure Vega doesn't display the controls in two places
+  .vega-bindings {
+    display: none !important;
+  }
+}
+
+.vgaVis__controls {
+  display: flex;
+  @include euiFontSizeS;
+
+  &:not(:empty) {
+    padding: $euiSizeS 0; // Adding a little bit of padding helps with the unnecessary scrollbars
+  }
+
+  &.vgaVis__controls--column {
+    flex-direction: column;
+  }
+
+  &.vgaVis__controls--row {
+    flex-direction: row;
+
+    > .vega-bind {
+      flex-grow: 1;
+    }
+  }
+
+  > .vega-bind {
+    .vega-bind-name {
+      display: inline-block;
+      width: $euiSizeM * 10 - $euiSize;
+    }
+
+    input[type="range"] {
+      width: $euiSizeM * 10;
+      display: inline-block;
+      vertical-align: middle;
+    }
+
+    label {
+      margin: 0 0 0 $euiSizeS;
+    }
+
+    select {
+      max-width: $euiSize * 10;
+    }
+
+    .vega-bind-radio label {
+      margin: 0 $euiSizeS 0 $euiSizeXS;
+    }
+  }
+}
+
+// Messages
+
+.vgaVis__messages {
+  position: absolute;
+  top: 0;
+  width: 100%;
+  margin: auto;
+  opacity: 0.8;
+  z-index: 1;
+  list-style: none;
+}
+
+.vgaVis__message {
+  margin: $euiSizeS;
+
+  .vgaVis__messageCode {
+    white-space: pre-wrap;
+    padding: $euiSizeS;
+  }
+}
+
+.vgaVis__message--warn .vgaVis__messageCode {
+  $backgroundColor: tintOrShade($euiColorWarning, 90%, 70%);
```

Comment:
Can we label these inner vars a little more specific. Just makes it easier to read. `$vgaVisBackgroundColor`...etc

Reply:
They're scoped to inside this selector the same way we calculate the background and text colors for callouts. I don't want it to look like they're possible to used anywhere else, so I at least renamed them to `$calculatedBackgroundColor`